src/vrct/vrctFs/config.go

```diff
@@ -0,0 +1,70 @@
+package vrctFs
+
+import "fmt"
+
+const (
+	JsonConfig = iota
+	YamlConfig
+	TomlConfig
+)
+
+type ConfigOption struct {
+	List   []ConfigOption          `bson:",omitempty"`
+	Map    map[string]ConfigOption `bson:",omitempty"`
+	IsNull bool                    `bson:",omitempty"`
+	Bool   *bool                   `bson:",omitempty"`
+	Int    *int                    `bson:",omitempty"`
+	Float  *float64                `bson:",omitempty"`
+	String *string                 `bson:",omitempty"`
+}
+
+func (c *ConfigOption) MergeWith(c2 ConfigOption) error {
+	if c2.IsNull {
+		return nil
+	}
+
+	if c2.Bool != nil {
+		if c.Bool != nil {
+			// TODO: use special struct for this kind of error
+			return fmt.Errorf("config conflict")
+		}
+		c.Bool = c2.Bool
+		return nil
+	}
+
+	if c2.Int != nil {
+		if c.Int != nil {
+			// TODO: use special struct for this kind of error
+			return fmt.Errorf("config conflict")
+		}
+		c.Int = c2.Int
+		return nil
+	}
+
+	if c2.Float != nil {
+		if c.Float != nil {
+			// TODO: use special struct for this kind of error
+			return fmt.Errorf("config conflict")
+		}
+		c.Float = c2.Float
+		return nil
+	}
+
+	if c2.String != nil {
+		if c.String != nil {
+			// TODO: use special struct for this kind of error
+			return fmt.Errorf("config conflict")
+		}
+		c.String = c2.String
+		return nil
+	}
```

Comment:
I would create the array of struct fields to check and loop over those struct fields using the "reflect" package in order to avoid repeated if statements.

Reply:
Actually I think I won't complete VRCT for config. There is still much other work with general VRCT and FsVRCT, so I think I will just create an issue for VRCT configs.
Especially that whole project is waiting for not read only rules (for merging this pr).